### PR TITLE
Enable Python3 tests and create first ZIP/TAR output archives.

### PR DESCRIPTION
**Pull request by bernhardkaindl**
@GeraldEV, @ashwin9390 - Update for the re-review:

1. Rebased to the updated master branch
2. Enabled the tests which were merged into master also on Python3

Ready for review again, but I can also split the commits if needed.

---
General overview:
- Cleanup/Refactor Github CI and use it to run Python2 and Python3 tests
- Add minimal mypy testing as soon as possible to avoid later mypy surprises
- Add more tests and make the tested bugtool code work for Python3 as well.

----
Among many others Python3 updates, this PR updates these functions and adds unit-tests for them:
- mdadm_arrays(),
- DBFilter.output()
- load_plugins()

#### Commit 1: Fixup GitHub Action Workflows and tests for Python3 tests

- Update `.github/workflows/main.yml` to be *ready* for running *Python3* unit tests in GitHub Actions

- Merge `.github/workflows/alpine-python2.yml` into `.github/workflows/main.yml`

- Update imports in `tests/integration` to be local imports as needed for static analysis checking of the tests.

### CODE REVIEW

**Review by ashwin9390 — APPROVED** [review]

File: tests/unit/test_fs_funcs.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    files = ["/proc/1/stat", "/proc/self/stat"]  # self/stat is more recent.
+    # return the files, ordered by their modification time, most recent first:
+    assert bugtool.get_recent_logs(files, 9) == files
+    assert bugtool.get_recent_logs([files[1], files[0]], 9) == files
+    # When verbosity is 1, only return the most recent file:
+    assert bugtool.get_recent_logs(files, 1) == [files[1]]
+
+
+def test_read_inventory(bugtool, dom0_template):
+    """Assert readKevValueFile() reading the a xensource inventory file"""
```
Comment (GeraldEV) [inline tests/unit/test_fs_funcs.py:16]: Key misspelled

File: tests/integration/conftest.py [hunk 2]
```diff
@@ -5,8 +5,8 @@
 
 import pytest
 
-from namespace_container import activate_private_test_namespace, mount, umount
-from utils import BUGTOOL_DOM0_TEMPL, BUGTOOL_OUTPUT_DIR, run
+from .namespace_container import activate_private_test_namespace, mount, umount
```
Comment (GeraldEV) [inline tests/integration/conftest.py:8]: Is this necessary for the actions? Seems like a step in the wrong direction for the python upgrade as relative imports won't be possible

Comment (bernhardkaindl) [reply tests/integration/conftest.py:8]: These are just local imports within the same Python test framework module.

Implicit relative imports have been deprecated in Python3, and `mypy` warned about them.

This converts these deprecated implicit module-local imports into explicit relative imports, which are of course not deprecated and of course needed and used by all Python3 modules.

So it converts Python2-style module-local imports into Python3-style module-local imports.

Here is of course a reference for anyone else who wants more examples and background:
https://realpython.com/absolute-vs-relative-python-imports/#relative-imports

File: tests/unit/test_fs_funcs.py (hunk 1 above)
Comment (bernhardkaindl) [reply tests/unit/test_fs_funcs.py]: Fixed, rebased to the updated master branch, added more fixes and enabled the merged tests.

Ready for review again, but I can also split the commits if needed.

**Review by GeraldEV — APPROVED** [review]

**Review by ashwin9390 — APPROVED** [review]

**Pull request merged**